File: library/src/main/java/org/xmtp/android/library/codecs/ReadReceiptCodec.kt
package org.xmtp.android.library.codecs

import com.google.protobuf.ByteString

val ContentTypeReadReceipt = ContentTypeIdBuilder.builderFromAuthorityId(
    "xmtp.org",
    "readReceipt",
    versionMajor = 1,
    versionMinor = 0,
)

object ReadReceipt

data class ReadReceiptCodec(override var contentType: ContentTypeId = ContentTypeReadReceipt) :
    ContentCodec<ReadReceipt> {

    override fun encode(content: ReadReceipt): EncodedContent {
        return EncodedContent.newBuilder().also {
            it.type = ContentTypeReadReceipt
            it.content = ByteString.EMPTY
        }.build()
    }

    override fun decode(content: EncodedContent): ReadReceipt {
        return ReadReceipt
    }

    override fun fallback(content: ReadReceipt): String? {
        return null
    }

    override fun shouldPush(content: ReadReceipt): Boolean = false
}

File: sdks/ios/Sources/XMTPiOS/Codecs/ReadReceiptCodec.swift
//
//  ReadReceiptCodec.swift
//
//
//  Created by Naomi Plasterer on 8/2/23.
//

import Foundation

public let ContentTypeReadReceipt = ContentTypeID(
	authorityID: "xmtp.org",
	typeID: "readReceipt",
	versionMajor: 1,
	versionMinor: 0,
)

public struct ReadReceipt {
	public init() {}
}

public struct ReadReceiptCodec: ContentCodec {
	public typealias T = ReadReceipt

	public init() {}

	public var contentType = ContentTypeReadReceipt

	public func encode(content _: ReadReceipt) throws -> EncodedContent {
		var encodedContent = EncodedContent()

		encodedContent.type = ContentTypeReadReceipt
		encodedContent.content = Data()

		return encodedContent
	}

	public func decode(content _: EncodedContent) throws -> ReadReceipt {
		ReadReceipt()
	}

	public func fallback(content _: ReadReceipt) throws -> String? {
		nil
	}

	public func shouldPush(content _: ReadReceipt) throws -> Bool {
		false
	}
}
